value tailLabel does not take parameters

Trying to compile scalastyle with dotty at commit: https://github.com/matthewfarwell/scalastyle/commit/92b81e0115bc70befeae98c86977282a26b686be

I get the compiler error:

```
error: value tailLabel624 does not take parameters
```

Running with -Ycheck:all and -Ycheck:firstTransform I get the following error:

```
[trace] Stack trace suppressed: run last compile:compileIncremental for the full output.
[error] (compile:compileIncremental) java.lang.AssertionError: assertion failed: error at /Users/ollie/dev/scalastyle/src/main/scala/org/scalastyle/Checker.scala:184
[error] type mismatch:
[error]  found   : Array[Class[_]]
[error]  required: Class[_]
[error] tree = Typed(JavaSeqLiteral(List(), TypeTree[RefinedType(TypeRef(ThisType(TypeRef(NoPrefix,lang)),Class), java$lang$Class$$T, TypeBounds(TypeRef(ThisType(TypeRef(NoPrefix,scala)),Nothing), TypeRef(ThisType(TypeRef(NoPrefix,scala)),Any)))]),TypeTree[RefinedType(TypeRef(ThisType(TypeRef(NoPrefix,scala)),Array), scala$Array$$T, TypeAlias(RefinedType(TypeRef(ThisType(TypeRef(NoPrefix,lang)),Class), java$lang$Class$$T, TypeBounds(TypeRef(ThisType(TypeRef(NoPrefix,scala)),Nothing), TypeRef(ThisType(TypeRef(NoPrefix,scala)),Any))), 0))])
```

linking to this line in Checker.scala:

```scala
      val c: Checker[_] = clazz.getConstructor().newInstance().asInstanceOf[Checker[_]]
```

Removing the block containing that line I get the original "taillabel does not take parameters" error instead of "type mismatch" when compiled with -Ycheck:all.

Any hints on how to better debug this?

--- a/src/dotty/tools/dotc/transform/TailRec.scala
+++ b/src/dotty/tools/dotc/transform/TailRec.scala
@@ -143,7 +143,11 @@ class TailRec extends MiniPhaseTransform with DenotTransformer with FullParamete
                     newOwners = label :: Nil
                   ).transform(rhsSemiTransformed)
                 })
-                Block(List(labelDef), ref(label).appliedToArgss(vparamss0.map(_.map(x=> ref(x.symbol)))))
+                val callIntoLabel = (
+                    if (dd.tparams.isEmpty) ref(label)
+                    else ref(label).appliedToTypes(dd.tparams.map(_.tpe))
+                  ).appliedToArgss(vparamss0.map(_.map(x=> ref(x.symbol))))
+                Block(List(labelDef), callIntoLabel)
             }} else {
               if (mandatory) ctx.error(
                 "TailRec optimisation not applicable, method not tail recursive",

--- a/tests/pos/tailcall/i1614.scala
+++ b/tests/pos/tailcall/i1614.scala
@@ -1,0 +1,9 @@
+object Foobar {
+  def apply(): Option[String] = { 
+    def foobar[A](f: (String, String) => A): List[A] = List[String]() match { 
+      case _ :: ls => foobar(f) 
+    } 
+    foobar((a1, a2) => (a1 + a2)).headOption 
+  } 
+}
+

